filament/src/details/Material.cpp

```diff
@@ -179,6 +194,45 @@ FMaterial::FMaterial(FEngine& engine, const Material::Builder& builder)
         mSubpassInfo.isValid = false;
     }
 
+    utils::FixedCapacityVector<MaterialConstant> constants;
+    // Older materials won't have a constants chunk, but that's okay.
+    parser->getConstants(&constants);
+
+    // Verify that all the constant specializations exist in the material and that their types match.
+    // The first two specialization constants are defined internally by Filament.
+    // Constants #2 onwards are user-defined in the material.
+    mSpecializationConstants.reserve(constants.size() + 2);
```

Comment:
should we keep some slots as reserved? Thinking 8 maybe?

Reply:
Yup, good idea.